tools/dev/dockerpkg.py

```diff
@@ -13,50 +13,40 @@
 """
 
 import logging
-import os
-import time
-import subprocess
-import sys
-import traceback
-from datetime import datetime
 from pathlib import Path
 
 import click
 
 log = logging.getLogger("dockerpkg")
 
 PHASE_1_INSTALL_TMPL = """
-ADD tools/c7n_{pkg}/pyproject.toml tools/c7n_{pkg}/poetry.lock /src/tools/c7n_{pkg}/
+ADD tools/c7n_{pkg}/pyproject.toml /src/tools/c7n_{pkg}/
 RUN if [[ " ${{providers[*]}} " =~ "{pkg}" ]]; then \
-    . /usr/local/bin/activate && \
-    cd tools/c7n_{pkg} && \
-    poetry install --without dev --no-root; \
+   uv sync --package c7n_{pkg} --frozen --inexact --no-install-workspace; \
```

Comment:
We use `--locked` during `make install`, does using `--frozen` instead here mean we expect/accept `uv.lock` to be out of sync during docker builds?

Reply:
re out of sync not during official builds, but possibly on local dev. 

more importantly its cause we're using partial installs to allow for provider selection, locked would complain in that context afaicr because its a different install set of packages with a different set of deps, where as frozen doesn't attempt to check that.